ensime_shared/ensime.py

```diff
@@ -944,9 +812,7 @@ def teardown(self):
 
     def current_client(self, quiet, bootstrap_server, create_client):
         """Return the current client for a given project."""
-        # Use current_file command because we cannot access self.vim
-        current_file_cmd = commands["current_file"]
-        current_file = self.vim.eval(current_file_cmd)
+        current_file = self.vim.current.buffer.name
```

Comment:
This is a method of the `Ensime` class, not `EnsimeClient`—we can discuss this more in the context of design discussion in the second part of these changes, but my reasoning so far is that using `vim` here is reasonable because the `Ensime` class basically represents *the Vim plugin itself* on the Python side, i.e. it's the base class for `NeovimEnsime` for instance.